crates/ra_ide/src/display/navigation_target.rs

```diff
@@ -416,3 +420,14 @@ pub(crate) fn description_from_symbol(db: &RootDatabase, symbol: &FileSymbol) ->
         }
     }
 }
+
+fn original_range(db: &RootDatabase, node: InFile<&SyntaxNode>) -> FileRange {
+    if let Some(range) = original_range_by_kind(db, node, OriginalRangeKind::CallToken) {
```

Comment:
I am not sure we need this more flexible `original_range_by_kind` function. I think the logic should be as follows:

* we try to map range back
* if we fail completely, we return the range of the macro_call
* if we land inside macro_argument, we return that range
* if we land inside macro_def, we return the range of the macro_call

Reply:
My reason for adding this function is, do `original_range` only used for `NavigationTarget` ? (And that's my reason to add `Any` even it is not used for now)

I have a feeling we would used it in other place, which we need to mapping the range inside `MacroDef`.

Reply:
Um.. Ah, I see what you mean right now, you mean we return the origin of what `original_source` find, and do the logic you mentioned above. It is functionally equal to my approach but much cleaner..